pageserver/src/tenant/timeline.rs

```diff
@@ -1100,6 +1102,19 @@ impl Timeline {
             return Ok(());
         }
 
+        match self.get_compaction_algorithm() {
+            CompactionAlgorithm::Tiered => self.compact_tiered(cancel, ctx).await,
+            CompactionAlgorithm::Legacy => self.compact_legacy(cancel, flags, ctx).await,
+        }
+    }
+
+    /// TODO: cancellation
```

Comment:
Note we should have error transformations in place already. I assume this is the background task cancellation token, which is ... less than important than `Timeline::cancel` or whatever the field was called.

Reply:
What would you recommend? Remove the argument plus todo, or remove the todo, or not do anything?

Reply:
My memory about us having merged some christians PR was wrong.

I think the argument is needless and will result in confusion, but I guess that is the fashionable choice. Let's just leave it like this.